features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/vitals/JankStatsActivityLifecycleListenerTest.kt

```diff
@@ -350,143 +345,78 @@ internal class JankStatsActivityLifecycleListenerTest {
         verifyNoInteractions(mockJankStatsProvider, mockJankStats, mockBundle)
     }
 
-    fun `M do nothing W onFrame() {zero ns duration}`(
-        @LongForgery timestampNs: Long,
-        @BoolForgery isJank: Boolean
-    ) {
-        // Given
-        val frameData = FrameData(timestampNs, 0L, isJank, emptyList())
-
-        // When
-        testedJankListener.onFrame(frameData)
-
-        // Then
-        verify(mockObserver, never()).onNewSample(any())
-    }
-
     @Test
-    fun `M forward frame rate to observer W doFrame() {acceptable frame rate}`(
-        @LongForgery timestampNs: Long,
-        @LongForgery(ONE_MILLISECOND_NS, ONE_SECOND_NS) frameDurationNs: Long,
-        @BoolForgery isJank: Boolean
-    ) {
+    fun `M do nothing W onActivityStarted() {android framework throws an exception}`() {
         // Given
-        val expectedFrameRate = (ONE_SECOND_NS.toDouble() / frameDurationNs.toDouble()).coerceAtMost(MAX_FPS)
-        val frameData = FrameData(timestampNs, frameDurationNs, isJank, emptyList())
+        whenever(mockWindow.addOnFrameMetricsAvailableListener(any(), any())) doThrow IllegalStateException()
 
         // When
-        testedJankListener.onFrame(frameData)
+        testedJankListener.onActivityStarted(mockActivity)
 
         // Then
-        verify(mockObserver).onNewSample(eq(expectedFrameRate, 0.0001))
+        // no-crash
```

Comment:
it is better to wrap then call to `onActivityStarted` in `assertDoesNotThrow`: in this case assertion intention will be clear and in case of the failure we will have a better message.

Same for the other similar tests.